wire-ios/Wire-iOS/Sources/UserInterface/E2EIdentity/E2eIFeatureChange.swift

```diff
@@ -0,0 +1,73 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireSyncEngine
+
+enum E2eIChangeAction: CaseIterable {
+
+    case getCertificate, remindLater
+
+    var title: String {
+        typealias Button = L10n.Localizable.FeatureConfig.Alert.MlsE2ei.Button
+
+        switch self {
+        case .getCertificate:
+            return Button.getCertificate
+        case .remindLater:
+            return Button.remindMeLater
+        }
+    }
+
+    var style: UIAlertAction.Style {
+        switch self {
+        case .getCertificate:
+            return .default
+        case .remindLater:
+            return .destructive
+        }
+    }
+
+    func action(_ handler: @escaping (E2eIChangeAction) -> Void) -> UIAlertAction {
+        return .init(title: title, style: style) { _ in handler(self) }
+    }
+
+}
```

Comment:
This feels a little bit over-engineered, wouldn't it be sufficient to craft the alert controller like:

```swift
let controller = ...
let getCertificateAction = UIAlertAction(L10n.Localizable..., style: .default) {
  // handle
}
let remindLaterAction = UIAlertAction(L10n.Localizable..., style: .destructive) {
  // handle
}

controller.addAction(getCertificateAction)
controller.addAction(remindLaterAction)

...
```